README.md

```diff
@@ -1,65 +1,169 @@
+{% extends "WebfactoryNavigationBundle:Navigation:navigation.html.twig" %}
 
-Wir haben uns darüber hinweggeholfen, indem die End-Projekte, die Navigationen mit webfactory/navigation-bundle
-ausgeben möchten, diese Ausgabe immer mittels eines standalone-embedded Controller durchführen mussten, dessen
-Ausgabe dann per ESI gecached werden konnte.
+{% block navigation_list %}
+    <nav class="project-specific-classes">
+        {{ parent() }}
+    </nav>
+{% endblock %}
+```    
 
-## webfactory/navigation-bundle Version 2.* ##
 
-Dieses Version fasst zunächst die Definition des Baumes und die Ausgabe einschließlich der Haken und Ösen (TM)
-zur Symfony-Integration in einem Bundle zusammen (keine externen Abhängigkeiten mehr), um das Leben in der Entwicklung
-zu vereinfachen, bis es mal einen guten Grund zur Aufteilung gibt.
+Modifying the NavigationTree
+----------------------------
 
-Der Ansatz, wie der Baum aufgebaut wird, ist neu:
+Implement a `Webfactory\Bundle\NavigationBundle\Build\BuildDirector`. Example:
 
-Es erfolgt (getrieben von einem Dispatcher) eine Breitensuche. Der aktuelle "Suchkontext" wird dann allen registrieren
-"TreeBuildern" präsentiert, die dann für den Kontext einen oder mehrere Knoten in den Baum einfügen können und/oder neue
-Kontexte zur weiteren Suche bilden.
+```php
+<?php
 
-Nach Abschluss der Suche ist der Baum fertig und wird mit Hilfe von Symfony ConfigCache gesichert. Er steht dann in
-jedem Request in der Form, wie er gebaut wurde, sofort zur Verfügung.
+namespace AppBundle\Navigation;
 
-Innerhalb des Requests wird dieser "Prototyp" dann noch modifiziert, z. B. um den aktiven Knoten zu markieren oder auch
-Unterknoten für dynamische Zustände (z. B. von Controllern aus) einzufügen.
+use JMS\ObjectRouting\ObjectRouter;
+use Symfony\Component\Config\Resource\FileResource;
+use Webfactory\Bundle\NavigationBundle\Build\BuildContext;
+use Webfactory\Bundle\NavigationBundle\Build\BuildDirector;
+use Webfactory\Bundle\NavigationBundle\Build\BuildDispatcher;
+use Webfactory\Bundle\NavigationBundle\Tree\Tree;
+use Webfactory\Bundle\WfdMetaBundle\Config\DoctrineEntityClassResource;
 
-Es ist ein Standard-Controller verfügbar, der den Baum als Navigation und als Breadcrumbs rendern kann (non-standalone),
-wobei das Theming anpassbar ist.
+final class KeyActionBuildDirector implements BuildDirector
+{
+    /** @var YourEntityRepository */
+    private $repository;
+
+    /** @var ObjectRouter */
+    private $objectRouter;
+
+    public function __construct(YourEntityRepository $repository, ObjectRouter $objectRouter)
+    {
+        $this->repository = $repository;
+        $this->objectRouter = $objectRouter;
+    }
+
+    public function build(BuildContext $context, Tree $tree, BuildDispatcher $dispatcher): void
+    {
+        if (!$this->isInterestedInContext($context)) {
+            return;
+        }
+
+        $this->addTreeCacheExpiryRule($dispatcher);
```

Comment:
"ExpiryRule" verspricht mehr, als dahinter steckt... Symfony's `ConfigCache` nennt die Dinger einfach `Resource`.

Reply:
Ich möchte hier die Intention deutlich machen, da scheint mir `Resource` zu vage. `addResourceForTreeCacheExpiry`? Oder habe ich da schon immer zu viel reininterpretiert - besserer Name?

Reply:
In Symfony heißen diese Methoden überall (Container, MessageCatalogue, Router...) einfach immer `addResource()`. Natürlich muss man wissen, was die `Resource` ist... aber ich weiß nicht, ob da der Methodenname ein hilfreicher clue ist?

Reply:
Die `addResource()`-Methode wird auf dem `Webfactory\Bundle\NavigationBundle\Build\BuildDispatcher` aufgerufen, da werden die Resources gesammelt um via `getResources()` abgerufen zu werden und zwar in der `\Webfactory\Bundle\NavigationBundle\Build\TreeFactory`, um da dann mit dem Baum zusammen in einen `\Symfony\Component\Config\ConfigCacheInterface` gesteckt zu werden. Ich finde diesen Weg arg weit, um schnell zu verstehen, dass hier Resources im Sinne des ConfigCacheInterface gemeint sind. Deshalb hab ich weiterhin das starke Bedürfnis, die `addResource()` als weiter innen liegende Interna in eine die Intention verdeutlichende `addTreeCacheExpiryRule()` (o.ä.) zu wrappen.